src/libraries/Common/src/Interop/Windows/SspiCli/SSPIWrapper.cs

```diff
@@ -246,11 +246,11 @@ public static bool QueryBlittableContextAttributes<T>(ISSPIInterface secModule,
                 contextAttribute == Interop.SspiCli.ContextAttribute.SECPKG_ATTR_CLIENT_SPECIFIED_TARGET);
 
 
-            Span<IntPtr> buffer = stackalloc IntPtr[1];
+            Span<byte> buffer = stackalloc byte[IntPtr.Size];
```

Comment:
The codegen for:
```C#
IntPtr p = default;
Span<byte> buffer = MemoryMarshal.AsBytes(new Span<IntPtr>(ref p));
```
ends up being better:
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4a9Jq049+ggJI8ZEMfKhKV6mpoDMTUgyEMaAbxoNPTe8RQMAsuQAFACULtReDO4RkV6yYthcADzAAJ4YMAB8DMAcAGZ5MgwAvAy4GNhgANbYkpIQYDnpMADaRhgAChhQLLK8AF4wALoaMbEMAHKmQbkFMiGjkQC+Hl6rnsQ+fv6koeGR0eOe8YlJ7V1Q2bOFUCVlFdW19Y3n3S3kI+uxU2JB/jD4aCpfzYKC4AAWtRYAEFcAAhZq4Gb5G4hBZfFZjfZeTZMba2PZfQ5HV63a5FUoAExgeWwHEkGEWRx+fwBQJBYMhkhh8MRQS4MAA7gwTslSZkgrA8jkUfM0UzPJjIl8Wv8MOCIJSDOJJKz1ZrtWJJAB5MR8CBcXAsKZGSS8Lj2gDmISGXzEUF4CmwGW0SDxk2mopSzSuiJCrkxSyAA==

Is there any way to write the equivalent without using MemoryMarshal.AsBytes? Maybe with some JIT tweaks?

Reply:
From my understanding, the codegen is better because your snippet doesn't require a GS cookie while we always emit it for stackallocs (whether we need it or not for such small stackallocs has been discussed here: https://github.com/dotnet/runtime/issues/52979)

Reply:
I'll see if we can safely remove the GS cookie when JIT sees that the stackalloc is consumed into Span